### PR TITLE
rbd: implement binding for rbd_mirror_image_instance_id_list

### PR DESCRIPTION
**Pull request by Rakshith-R**
This commit adds MirrorImageInstanceIDList and
MirrorImageInstanceIDIter with necessary helper functions
and tests.

Fixes: #483

Signed-off-by: Rakshith R <rar@redhat.com>

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code

### CODE REVIEW

File: rbd/mirror_test.go [hunk 1]
```diff
@@ -971,3 +971,135 @@ func TestMirrorImageLists(t *testing.T) {
...
+	ioctx2, err := conn2.OpenIOContext(poolName)
+	assert.NoError(t, err)
+	defer func() {
+		ioctx2.Destroy()
+	}()
+
+	err = SetMirrorMode(ioctx2, MirrorModeImage)
+	require.NoError(t, err)
+
+	fmt.Printf("TOKEN: %s\n", string(token))
```
Comment (phlogistonjohn) [inline rbd/mirror_test.go:1036]: Leftover debugging?

**Review by phlogistonjohn — COMMENTED** [review]
Wow! What a nice surprise to see today. :-)  On quick skim it looks pretty good. I'll do a 2nd more thorough review soon.

File: rbd/mirror_test.go (hunk 1 above)
Comment (Rakshith-R) [reply rbd/mirror_test.go]: yes, removed it now :sweat_smile:

**Review by phlogistonjohn — APPROVED** [review]
LGTM, thanks!

**Review by nixpanic — APPROVED** [review]

**Pull request merged**